### PR TITLE
feat: add Responder.{SetContentLength,HeaderAdd,HeaderSet} methods

### PR DESCRIPTION
**Pull request by maxatome**
Another way to solve the problem reported by @vrnvu in #137, and also #20.

### CODE REVIEW

File: match.go [hunk 1]
```diff
@@ -500,3 +500,9 @@ func (b *bodyCopyOnRead) Read(p []byte) (n int, err error) {
 func (b *bodyCopyOnRead) Close() error {
 	return nil
 }
+
+// Len returns the buffer totzal length, whatever the Read position in body is.
```
Comment (c-roussel) [inline match.go]: totzal => total

File: response_test.go [hunk 2]
```diff
@@ -586,6 +587,171 @@ func TestResponder_Then(t *testing.T) {
...
+
+	orig := NewStringResponder(200, "body")
+	origNilHeader := ResponderFromResponse(&http.Response{
+		Status:        "200",
+		StatusCode:    200,
+		Body:          NewRespBodyFromString("body"),
+		ContentLength: -1,
+	})
+
+	// until go1.17, http.Header cannot contains nil values after a Header.Clone()
```
Comment (c-roussel) [inline response_test.go]: contains => contain

File: response.go [hunk 3]
```diff
@@ -217,6 +217,153 @@ func (r Responder) Then(next Responder) (x Responder) {
...
+// See also [Responder.HeaderSet] and [Responder.SetContentLength].
+func (r Responder) HeaderAdd(h http.Header) Responder {
+	return func(req *http.Request) (*http.Response, error) {
+		resp, err := r(req)
+		if err != nil {
+			return nil, err
+		}
+		nr := *resp
+		if nr.Header == nil {
+			nr.Header = make(http.Header, len(h))
```
Comment (c-roussel) [inline response.go:320]: this could be nr.Header = h.Clone() and return

Comment (maxatome) [reply response.go:320]: if we do that we don't CanonicalHeaderKey'ize each key is this case

Comment (c-roussel) [reply response.go:320]: Fine with me

**Review by c-roussel — APPROVED** [review]

**Pull request merged**